### PR TITLE
[test] [ir] [bug] Fix googletest on Windows and fix a bug in IR Builder

### PR DESCRIPTION
**Pull request by xumingkuan**
Related issue = #2225 #2201 #2193

2 minor fixes in 1 PR :)

Reference: https://github.com/google/googletest/blob/master/googletest/README.md

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

**Review by BillXu2000 — APPROVED** [review]
LGTM

File: taichi/ir/ir_builder.cpp [hunk 1]
```diff
@@ -15,7 +15,7 @@ void IRBuilder::reset() {
 }
 
 std::unique_ptr<IRNode> IRBuilder::extract_ir() {
-  auto &&result = std::move(root_);
+  auto result = std::move(root_);
   reset();
   return std::move(result);
```
Comment (BillXu2000) [inline taichi/ir/ir_builder.cpp:20]: Maybe remove std::move?

Comment (k-ye) [inline taichi/ir/ir_builder.cpp:20]:
```suggestion
  return result;
```

Copy elision: https://stackoverflow.com/questions/4316727/returning-unique-ptr-from-functions

**Review by k-ye — APPROVED** [review]
Thank you for fixing Windows!

**Pull request merged**